src/WorkOS.net/Services/Webhooks/WebhookService.cs

```diff
@@ -0,0 +1,219 @@
+namespace WorkOS
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using Newtonsoft.Json;
+
+    public class WebhookService
+    {
+        private const int DefaultTimeTolerance = 300;
+
+        public static bool ConstantTimeAreEqual(byte[] a, byte[] b)
+        {
+            int i = a.Length;
+            if (i != b.Length)
+            {
+                return false;
+            }
+
+            int cmp = 0;
+            while (i != 0)
+            {
+                --i;
+                cmp |= a[i] ^ b[i];
+            }
+
+            return cmp == 0;
+        }
+
+        /// <summary>
+        /// Parses a JSON string from a webhook, verifies and deseralizes into a Webhook Object.
+        /// </summary>
+        /// <param name="json">The JSON string to parse.</param>
+        /// <param name="signatureHeader">
+        /// The value of the header from the webhook request.
+        /// </param>
+        /// <param name="secret">The webhook endpoint's signing secret.</param>
+        /// <param name="tolerance">The time tolerance, in seconds.</param>
+        /// <returns>The deserialized JSON.</returns>
+        public Webhook ConstructEvent(
+            string json,
+            string signatureHeader,
+            string secret,
+            long tolerance)
+        {
+            if (this.VerifyHeader(json, signatureHeader, secret, tolerance))
+            {
+                Webhook webhookVerified = JsonConvert.DeserializeObject<Webhook>(json);
+                return webhookVerified;
+            }
+            else
+            {
+                throw new Exception("Unable to verify Webhook");
+            }
+        }
+
+        /// <summary>
+        /// Verify Header.
+        /// </summary>
+        /// <param name="payload">
+        /// json payload.
+        /// </param>
+        /// <param name="sigHeader">
+        /// signature header of webhook.
+        /// </param>
+        /// <param name="secret">
+        /// secret.
+        /// </param>
+        /// <param name="tolerance">
+        /// time tolerance for timing attacks.
+        /// </param>
+        /// <returns>Boolean True or False of verification.</returns>
+        public bool VerifyHeader(string payload, string sigHeader, string secret, long tolerance)
+        {
+            var timeAndSignature = this.GetTimestampAndSignature(sigHeader);
+            var timeStamp = timeAndSignature.Item1;
+
+            if (!this.Verify_Time_Tolerance(timeStamp, tolerance))
+            {
+                throw new Exception("Timestamp outside of the tolerance zone");
+            }
+
+            string signatureHash = timeAndSignature.Item2;
```

Comment:
There's some more places where we can use `var` instead.

You pretty much never want to use an explicit type variable for local bindings if it can be inferred.

Reply:
Think I got the last of them